### PR TITLE
[3D] [point cloud] adjust screen space error from UI

### PR DESCRIPTION
**Pull request by NEDJIMAbelgacem**
## Description
This adds the ability to choose the screen space error used to render the point cloud.
screen_space_error_adjustment

### CODE REVIEW

File: src/3d/qgspointcloudlayer3drenderer.cpp [hunk 1]
```diff
@@ -148,3 +148,13 @@ void QgsPointCloudLayer3DRenderer::resolveReferences( const QgsProject &project
 {
   mLayerRef.setLayer( project.mapLayer( mLayerRef.layerId ) );
 }
+
+double QgsPointCloudLayer3DRenderer::maximumScreenError() const
```
Comment (nyalldawson) [inline src/3d/qgspointcloudlayer3drenderer.cpp:152]: This should will need to be saved/restored from XML too

File: src/app/3d/qgspointcloudlayer3drendererwidget.cpp [hunk 2]
```diff
@@ -50,6 +50,7 @@ QgsPointCloudLayer3DRenderer *QgsPointCloudLayer3DRendererWidget::renderer()
   QgsPointCloud3DSymbol *sym = mWidgetPointCloudSymbol->symbol();
   renderer->setSymbol( sym );
   renderer->setLayer( qobject_cast<QgsPointCloudLayer *>( mLayer ) );
+  renderer->setMaximumScreenError( mWidgetPointCloudSymbol->maximumScreenError() );
```
Comment (wonder-sk) [inline src/app/3d/qgspointcloudlayer3drendererwidget.cpp:53]: There should be also some code to initialize the GUI from the maximum screen error set already set in renderer...

**Comment by NEDJIMAbelgacem** [comment]
Sorry for forgetting about that @wonder-sk @nyalldawson

**Review by nyalldawson — APPROVED** [review]
Looks good to me!

I think we should reconsider how we expose these settings to users in both the 2d and 3d widgets as the currently phrasing is quite technical. I'm thinking of rebranding them as "Point thinning" or "Rendering quality" instead. But I'll chat with @nirvn about this and address as a follow up PR.

**Comment by nyalldawson** [comment]
(@NEDJIMAbelgacem @wonder-sk we may also want to rethink the default errors we set here. Possibly these should be dependant on the actual size of the point cloud, so for smaller clouds (ie smaller numbers of points) we default to a lower value vs larger clouds...)

**Comment by wonder-sk** [comment]
@NEDJIMAbelgacem can we change the default screen error to 1 pixel instead of 5px (that number I left there originally only to clearly see how the data are being loaded/unloaded)

@nyalldawson sure we can change the naming, maybe it could be just some slider "dense' -> "sparse" (or "aggresive" -> "conservative") ... this same option is also available for terrain configuration - so maybe it would be good to choose a term that is not specific to point cloud data.

**Pull request merged**